### ISSUE TITLE
Override inherited field's @Schema annotation in a subclass?

### ISSUE THREAD

**Issue by retrofreak83** [swagger-api/swagger-core#3688]
I am working on a Spring REST API, documented using OpenAPI and SpringDoc. I have a DTO class for generic responses (`ResponseDTO`) and an extended one for error messages extending the first one (`ErrorResponseDTO`).

```
public class ResponseDTO
{
  @Schema(description="This field mostly equals SUCCESS")
  private final String type;
}
```

```
public class ErrorResponseDTO extends ResponseDTO
{
  @Schema(description="this field contains a detailed error message")
  private final String errorDetails;
}
```

Now the class `ErrorResponseDTO` inherits the type field. And naturally, type in `ErrorResponseDTO` shall have other / additional values than in `ResponseDTO`.

The problem now is that in Swagger, the description of `ErrorResponseDTO` inherits the description of the field type from `ResponseDTO`.

My question is: Is there a possibility to override the field's description in `ErrorResponseDTO`?

**Comment by bogomolov-a-a** [comment]
@retrofreak83,  I merge field @Schema annotation member values (title,description and others) in custom ModelConverter.
Logic is very simplify: if sibling override description(for example), I say ModelConverter - update, please in target field(where use this type). But I use only getter in child class. I.e. I describe field errorDescription in type ErrorResponseDTO with his getter.

**Comment by retrofreak83** [comment]
What do you mean by "custom ModelConverter"? I don't see this being an officially supported concept of SpringDoc, at least the documentation does not talk about it.

However, I noticed that when I annotate getter of super class and sibling class instead of the field, the documentation in OpenAPI is overridden correctly.

**Comment by Akaame** [comment]
@retrofreak83 

Your solution to this problem is:

```java
public class ResponseDTO
{
  @Schema(description="This field mostly equals SUCCESS")
  private final String type;
}
```

```java
public class ErrorResponseDTO extends ResponseDTO
{
  @Schema(description="this field contains a detailed error message")
  private final String errorDetails;
  
  @Schema(description="This field is ERROR")
  public String getType() {
    return type;
  }
}
```

Could you please confirm?

Cheers.

**Comment by retrofreak83** [comment]
Not quite. In `ResponseDTO`, I would also annotate the getter then instead of the field (AFAIR).

**Comment by Akaame** [comment]
Hey thanks for the quick response. Turns out that does not make a difference so the solution (annotate field in base class and getter in derived) also works.

Cheers.

**Comment by nkavian** [comment]
That does not work for me. In my case, instead of a base class, I put the `@Schema` annotation on an enum class. Now in the DTO class, I can't override the description.

**Comment by MichakrawSB** [comment]
Closing this issue as part of our [housekeeping & cleanup](https://github.com/swagger-api/swagger-core/issues/5074).
This issue is outdated and/or inactive for a long time, and we’re not able to revisit it meaningfully.

If this problem is still relevant, please open a new issue and reference this one for context.
Thank you for your understanding!
